scalafix-core/src/main/scala/scalafix/internal/patch/ReplaceSymbolOps.scala

```diff
@@ -126,10 +143,20 @@ object ReplaceSymbolOps {
             if sig.name != parent.value =>
           Patch.empty // do nothing because it was a renamed symbol
         case Some(parent) =>
+          val causesCollision = getGlobalImports(ctx.tree).exists { importStmt =>
```

Comment:
It would be more performant to get the global imports from the start and manipulate them as patches are processed. We want to handle for `foo.Bar => foo.Baz` and `bar.Foo => bar.Baz` without breaking compilation, so we have to consider previous renames as well. Now that I think more on it, I guess for completeness we'd also want to handle non-import collisions. Open to suggestions on these.

Reply:
> Now that I think more on it, I guess for completeness we'd also want to handle non-import collisions. Open to suggestions on these.

Indeed, although this would be more complex as  there is not enough information in the tree nor in semanticdb to list all locals for example. You'd have to start a presentation compiler instance (outside the scalafix API), just like [ExplicitResultTypes](https://github.com/scalacenter/scalafix/blob/e11610fb2abcf78355c57794b04626e2d2c5c24e/scalafix-rules/src/main/scala-2/scalafix/internal/rule/CompilerTypePrinter.scala), which would greatly increase code complexity and ease-of-use (as scalafix's Scala binary version must match the target's). Without going that way, there are some heuristics that could prevent non-import collisions (getting visibile members from super traits, getting local val names via the tree, etc), but it's a lot of work to cover them all.

Reply:
> It would be more performant to get the global imports from the start

Indeed - is it possible to call `getGlobalImports` once outside the tree traversal?